[Improve] code style and code format

### Search before asking

- [x] I searched in the [issues](https://github.com/fast-excel/fastexcel/issues) and found nothing similar.


### Motivation

I noticed that `style/checkstyle/fastexcel_checkstyle.xml` already exists, but `Checkstyle` is not used in the project build, and there are many formatting errors and optimization issues in the current project code. I suggest replacing it with `Spotless`, or using both together to ensure that formatting errors are automatically corrected during development and `CI`.

### Solution

- use `maven-checkstyle-plugin`
- With `Spotless`, we could fix the legacy formatting errors once for all and developers could fix new format errors with a single line of command `mvn spotless:apply`.

### Alternatives

_No response_

### Anything else?

<img width="870" height="852" alt="Image" src="https://github.com/user-attachments/assets/a5a78453-5e69-4c80-a03c-af9fbd7c5365" />

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

> > It won't resolve the java module issue as below screenshot
> 
> Could you provide the full error log or a screenshot for how to reproduce it?

enable test execution during mvn test phase, then trigger mvn test
<maven.test.skip>false</maven.test.skip>

for core test , it will be failed by below case. still not run the full test (hours level with demo and temp package)

[ERROR] Errors: 
[ERROR]   CacheDataTest.t02ReadAndWriteInvoke:90 » InaccessibleObject Unable to make field private final java.util.Map sun.reflect.annotation.AnnotationInvocationHandler.memberValues accessible: module java.base does not "opens sun.reflect.annotation" to unnamed module @441772e
[ERROR]   CacheDataTest.t03ReadAndWriteInvokeMemory:146 » InaccessibleObject Unable to make field private final java.util.Map sun.reflect.annotation.AnnotationInvocationHandler.memberValues accessible: module java.base does not "opens sun.reflect.annotation" to unnamed module @441772e

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -58,6 +58,10 @@ FastExcel 欢迎社区的每一位用户和开发者成为贡献者。无论是�
 
 ### 工作区准备
 
+开发 FastExcel 需要 Java 工具链。目前，开发工具链要求 JDK 21 及以上的版本，但在编码过程中必须使用 JRE 1.8 兼容的语言特性，保证产出的 JAR 包能在 JRE 1.8 及以上版本环境中运行。
+
+您可以使用 [SDKMAN](https://sdkman.io/) 等工具配置多版本的 Java 工具链。如果使用 IntelliJ IDEA 开发，可以设置项目 [Language Level](https://www.jetbrains.com/help/idea/project-settings-and-structure.html#language-level) 为 8 以确保后向兼容。
+
 确保您已注册 GitHub 账号，并按照以下步骤完成本地开发环境配置：
 
 1. **Fork 仓库**：在 FastExcel 的 [GitHub 页面](https://github.com/fast-excel/fastexcel) 点击 `Fork` 按钮，将项目复制到您的 GitHub 账户下，例如：`https://github.com/<your-username>/fastexcel`。

